### PR TITLE
change profile happy path e2e

### PR DESCRIPTION
**Pull request by Meyclem**
## Description

This PR is a first step in the improvement of our e2e tests flows.
I mostly worked on the "profile and addresses happy path" test and added some tooling.

## Type of change

- [x] **Chore** (non-breaking change which refactors / improves the existing code base).
- [ ] **Bug fix** (non-breaking change which fixes an issue).
- [ ] **New feature** (non-breaking change which adds functionality).
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## Issues

[CU-9u11zz](https://app.clickup.com/t/9u11zz)

## Checklist

- [x] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

**Comment by frieda-ewlines** [comment]
Task linked: [CU-9u11zz Improve the way of running e2e tests in local env](https://app.clickup.com/t/9u11zz)

File: .env_vars.sample
Comment (Meyclem) [inline .env_vars.sample:51]: I added a script to delete the test user from Connect.Profile's database and we need that to be authorized on the targeted endpoint.

File: bin/e2e/delete-profile-user.ts
Comment (Meyclem) [inline bin/e2e/delete-profile-user.ts:28]: The user may not exist, mainly for two reasons:
- we are testing against the mock server
- It has already been deleted
When it is the case, we don't want the script to fail.

File: package.json
Comment (Meyclem) [inline package.json:17]: The `deleteProfileUser` script is started each time we run the e2e tests in local.
It avoids us having to manually do it, either directly in the Profile's database, either with a call to the admin endpoint.

Let me know if you prefer to run it separately.

File: tests/e2e/utils/authenticate-to-connect.ts
Comment (Meyclem) [inline tests/e2e/utils/authenticate-to-connect.ts:18]: When the user is already logged in there's the "Continue" button instead of the login form.
It happens on the Profile happy path since I separated profile and addresses tests (in the same file), so I added it in this function.

File: package.json
Comment (Fuzznimp) [reply package.json:17]: I'm fine with this 👍.

File: bin/e2e/delete-profile-user.ts
Comment (Oluane) [reply bin/e2e/delete-profile-user.ts:28]: We already had this script (https://github.com/fewlinesco/connect-account/blob/main/bin/e2e/cleaning-profile-staging.ts), which is used in `config.yml` (l70)

File: tests/e2e/utils/authenticate-to-connect.ts
Comment (Fuzznimp) [reply tests/e2e/utils/authenticate-to-connect.ts:18]: We'll need to remember this when we'll refactor the auth middleware to redirect to `/account` from `/` when the user is already logged in.

**Review by Fuzznimp — COMMENTED** [review]
Good for me 👍.

File: bin/e2e/delete-profile-user.ts
Comment (Meyclem) [reply bin/e2e/delete-profile-user.ts:28]: Yes, but since it's tailored for staging I preferred to add it only for e2e.
Do you prefer me to tweak this and use the script I wrote in the code you shared to avoid repetition?

Comment (Oluane) [reply bin/e2e/delete-profile-user.ts:28]: It isn't really tied to staging, there's just the API_KEY env var to change and it could be used anywhere I think. The script name could also be changed, to not reference staging. Anyway, do as you want, but we don't need to have the both imo.

Comment (Meyclem) [reply bin/e2e/delete-profile-user.ts]: I changed the code:
- removed my function
- tweaked a bit the existing one to be less specific to staging (mainly naming).

I also changed the env variable name and added it to CircleCi (so it is there 2 time for now with two different names but I'll delete the `CONNECT_PROFILE_STAGING_API_KEY` one just before the merge of this PR).

File: tests/e2e/profile-happy-path.test.ts
Comment (Oluane) [inline tests/e2e/profile-happy-path.test.ts:68]: Is it intended ?

File: tests/e2e/profile-happy-path.test.ts
Comment (Oluane) [inline tests/e2e/profile-happy-path.test.ts:88]: Why did you remove this except ? It was meant to verify that the add profile flow worked as expected, since if it is successful, we are redirected to the Profile overview.

File: tests/e2e/utils/authenticate-to-connect.ts
Comment (Oluane) [reply tests/e2e/utils/authenticate-to-connect.ts:18]: It could be great for consistency to update `fill-incorrect-identity-input-values.test.ts`, which is the only other test file that has two separated test cases, and make it use what you did with `profile-happy-path.test.ts` and `authenticate-to-connect.ts`

Comment (Meyclem) [reply tests/e2e/utils/authenticate-to-connect.ts:18]: Done 👍

File: tests/e2e/profile-happy-path.test.ts
Comment (Meyclem) [reply tests/e2e/profile-happy-path.test.ts:88]: It is useless in fact, because if we aren't on that page, the line 👇  will fail and will also make the test fail.
```ts
await click(localizedStrings.profileOverview.updateInfo);
```

At first I forgot to put it back but it doesn't matter so I didn't put it back 😁

File: tests/e2e/profile-happy-path.test.ts
Comment (Meyclem) [reply tests/e2e/profile-happy-path.test.ts:68]: Absolutely not.

File: tests/e2e/fill-incorrect-identity-input-values.test.ts
Comment (Fenntasy) [inline tests/e2e/fill-incorrect-identity-input-values.test.ts:51]: why?

Comment (Meyclem) [reply tests/e2e/fill-incorrect-identity-input-values.test.ts:51]: Two reasons:
- For now, the function `authenticateToConnect` has some `expect` inside and we must add them in this count for the test to pass. I find that it make it useless because not reflecting the number of "real assertions" in the test.
- It's useful when something can pass or not, like with errors or calls or whatever that can "silently" fail, but here, we expect some things that will make the test fail if there are not there anyway. For example, we expect some input before trying to write inside. If the input is not there, the write action will fail and make the test fail in the same time.

Comment (Fenntasy) [reply tests/e2e/fill-incorrect-identity-input-values.test.ts:51]: > It is an absolute hell to try our test suites with expect.assertions. And I'll find a better way to handle this but it's not for this PR

I'm curious on this point.

I would find it better to remove the expects from the `authenticateToConnect` since we don't need to test that though.

Comment (Meyclem) [reply tests/e2e/fill-incorrect-identity-input-values.test.ts:51]: > It is an absolute hell to try our test suites with expect.assertions

Not talking about code here but we have many `expect`, and as I "play" with the tests and move things, I had to keep the count in mind and it was pretty annoying.

> I would find it better to remove the expects from the authenticateToConnect

I tried to not do that in this PR to keep changes  at a minimum because if I do, I have to change every `expect.assertions` for e2e tests.

As you prefer!

Comment (Fenntasy) [reply tests/e2e/fill-incorrect-identity-input-values.test.ts:51]: > Not talking about code here but we have many expect, and as I "play" with the tests and move things, I had to keep the count in mind and it was pretty annoying.

You definition of "absolute hell" is quite a gentle one 😅 . Just comment the line while you're doing your tests and add it back at the end :)
But it's really a good practice to have it,I think it can save our beacon even in e2e tests.

> I tried to not do that in this PR to keep changes at a minimum because if I do, I have to change every expect.assertions for e2e tests. But can do, it's as you prefer!

Well you have my blessing 😇

**Review by Fenntasy — APPROVED** [review]

**Pull request merged**
